sscanss/app/dialogs/preferences.py

```diff
@@ -255,51 +255,49 @@ def graphicsForm(self):
-        combo_box.setCurrentIndex(size.index(value) if value in size else 0)
-        combo_box.currentIndexChanged.connect(lambda i, v=size: self.changeSetting(v[i]))
-        layout.addWidget(combo_box)
-        layout.addStretch(1)
+        fiducial = widgets.SliderTextInput(self, value, settings.default(key).limits)
+        fiducial.slider.setProperty(self.prop_name, (key, value))
+        fiducial.slider.valueChanged.connect(lambda: self.changeSetting(fiducial.slider.value()))
+        layout.addWidget(fiducial.slider)
+        layout.addWidget(fiducial.slider_value)
+        layout.setStretch(1, 3)
+        layout.setStretch(2, 1)
         main_layout.addLayout(layout)
         main_layout.addSpacing(5)
 
         layout = QtWidgets.QHBoxLayout()
         key = settings.Key.Measurement_Size
         value = settings.value(key)
         layout.addWidget(QtWidgets.QLabel('Measurement Points:'))
-        size = (5, 15, 30)
-        combo_box = QtWidgets.QComboBox()
-        combo_box.addItems(['Small', 'Medium', 'Large'])
-        combo_box.setProperty(self.prop_name, (key, value))
-        combo_box.setCurrentIndex(size.index(value) if value in size else 0)
-        combo_box.currentIndexChanged.connect(lambda i, v=size: self.changeSetting(v[i]))
-        layout.addWidget(combo_box)
-        layout.addStretch(1)
+        measurement_points = widgets.SliderTextInput(self, value, settings.default(key).limits)
+        measurement_points.slider.setProperty(self.prop_name, (key, value))
+        measurement_points.slider.valueChanged.connect(lambda: self.changeSetting(measurement_points.slider.value()))
+        layout.addWidget(measurement_points.slider)
+        layout.addWidget(measurement_points.slider_value)
+        layout.setStretch(1, 3)
+        layout.setStretch(2, 1)
         main_layout.addLayout(layout)
         main_layout.addSpacing(5)
 
         layout = QtWidgets.QHBoxLayout()
         key = settings.Key.Vector_Size
         value = settings.value(key)
         layout.addWidget(QtWidgets.QLabel('Measurement Vectors:'))
-        size = (10, 35, 50)
-        combo_box = QtWidgets.QComboBox()
-        combo_box.addItems(['Small', 'Medium', 'Large'])
-        combo_box.setProperty(self.prop_name, (key, value))
-        combo_box.setCurrentIndex(size.index(value) if value in size else 0)
-        combo_box.currentIndexChanged.connect(lambda i, v=size: self.changeSetting(v[i]))
-        layout.addWidget(combo_box)
-        layout.addStretch(1)
+        measurement_vectors = widgets.SliderTextInput(self, value, settings.default(key).limits)
+        measurement_vectors.slider.setProperty(self.prop_name, (key, value))
+        measurement_vectors.slider.valueChanged.connect(lambda: self.changeSetting(measurement_vectors.slider.value()))
+        layout.addWidget(measurement_vectors.slider)
+        layout.addWidget(measurement_vectors.slider_value)
+        layout.setStretch(1, 3)
```

Comment:
Please layout the `SliderTextInput` inside the widget itself the add a stretch i.e. `layout.addStretch(1)`. Look at the examples below that use the `ColourPicker`